### PR TITLE
FLS Callbacks

### PR DESCRIPTION
**Pull request by learn-more**
FLS implementation passes all tests.

Test passes from xp up to 10.
https://testbot.winehq.org/JobDetails.pl?Key=38266

### CODE REVIEW

**Comment by HBelusca** [comment]
Is there a possibility to check whether these callbacks are called under SEH on Windows?

**Comment by Doug-Lyons** [comment]
Regarding SEH if you look at this MSDN example here:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms686991(v=vs.85).aspx
Since there is a programmed ErrorExit wouldn't this indicate that it is not called inside SEH blocks?

**Comment by learn-more** [comment]
@Doug-Lyons I fail to see how that relates to SEH.
@HBelusca I want to focus on getting this to work first, since I see the logging a lot.
After that, we can check to see if we can improve it.

File: dll/win32/kernel32/client/fiber.c [hunk 1]
```diff
@@ -29,7 +31,28 @@ VOID
...
+    PFLS_CALLBACK_FUNCTION lpCallback;
+
+    pFlsData = FlsData;
+
+    RtlAcquirePebLock();
+    FlsHighIndex = NtCurrentPeb()->FlsHighIndex;
+    RemoveEntryList(&pFlsData->ListEntry);
+    RtlReleasePebLock();
+
+
```
Comment (HBelusca) [inline dll/win32/kernel32/client/fiber.c]: extra newline

**Comment by learn-more** [comment]
At this point checking for SEH makes sense, since we hold the lock.

File: dll/win32/kernel32/client/fiber.c (hunk 1 above)
Comment (sanchaez) [reply dll/win32/kernel32/client/fiber.c]: No shit, Sherlock!

File: modules/rostests/apitests/kernel32/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ list(APPEND SOURCE
     dosdev.c
     FindActCtxSectionStringW.c
     FindFiles.c
+    FLS.c
```
Comment (encodedpr) [inline modules/rostests/apitests/kernel32/CMakeLists.txt]: Upper case filename

**Pull request merged**